extension/js/common/helpers.ts

```diff
@@ -63,38 +71,49 @@ const parseAndCheckPrivateKeys = async (decryptedPrivateKeys: string[]) => {
 };
 
 const filterKeysToSave = async (candidateKeys: Key[], existingKeys: KeyInfoWithIdentity[]) => {
+  // todo: check for uniqueness of candidateKeys identities here?
   if (!existingKeys.length) {
-    return candidateKeys;
+    return { keysToRetain: [], unencryptedKeysToSave: candidateKeys };
   }
-  const result: Key[] = [];
+  const keysToRetain: Key[] = [];
+  const unencryptedKeysToSave: Key[] = [];
   for (const candidate of candidateKeys) {
     const longid = KeyUtil.getPrimaryLongid(candidate);
     const keyToUpdate = existingKeys.filter(ki => ki.longid === longid && ki.family === candidate.family);
```

Comment:
Call this `existingKey`, and instead of a `filter` let's do `find` or similar that yields undefined or the result, not an array. We don't need the `length > 1` check here, since this comes from our own storage.